octoprint_mrbeam/gcodegenerator/img_separator.py

```diff
@@ -340,9 +340,7 @@ def _get_contours(self, img, method=cv2.RETR_EXTERNAL):
             )
 
         else:
-            _, contours, hierarchy = cv2.findContours(
-                img, method, cv2.CHAIN_APPROX_SIMPLE
-            )
+            contours, hierarchy = cv2.findContours(img, method, cv2.CHAIN_APPROX_SIMPLE)
```

Comment:
Teja:

> das "_, " am anfang der Zeile ist wichtig, da opencv in verschiedenen versionen da unterschiedlich viele parameter zurück gibt. genau deshalb ist überhaupt diese versionsweiche da - und axel hat in seinem commit in beiden wegen den gleichen code.
```suggestion
            _, contours, hierarchy = cv2.findContours(img, method, cv2.CHAIN_APPROX_SIMPLE)
```

Reply:
@amrsoll Please also check other OPenCV call, since OpenCV holds more fund thing like this. according to teja.

Reply:
The underscore "variable" `_` just ignores the given returned element, but it is not a glob that catches multiple elements of an iterable.

```python
_, a = 1, 2, 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
>>> _, a, b  = 1, 2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 3, got 2)
```
The version of OpenCV that is commonly found on our devices only returns 2 elements, so either we:

- make a case-ish statement for different versions of OpenCV
- Freeze the OpenCV version

but we can't have it both ways

Reply:
Also I believe that this branch might be stale afte all the changes that Teja introduced in the other branch :)

Reply:
There is already a case-ish statement for different versions of openCV. Your removal of the _ in the second part broke the code @amrsoll

Reply:
ah yes you are right I didn't see

Reply:
Then fine, but shouldn't we be deleting references to older opencv code? Aren't all of the devices at least on cv2 version 3.4 ?